sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala

```diff
@@ -610,24 +610,15 @@ abstract class TreeNode[BaseType <: TreeNode[BaseType]] extends Product with Tre
 
   /**
    * Returns a copy of this node where `f` has been applied to all the nodes in `children`.
+   * This method force making a copy of the nodes even if no child has been changed.
    * @param f The transform function to be applied on applicable `TreeNode` elements.
-   * @param forceCopy Whether to force making a copy of the nodes even if no child has been changed.
    */
-  private def mapChildren(
-      f: BaseType => BaseType,
-      forceCopy: Boolean): BaseType = {
-    var changed = false
+  private def mapChildrenWithForceCopy(f: BaseType => BaseType): BaseType = {
```

Comment:
Is there a more suitable method name?

Reply:
Should we move this method into `clone()` method? It is now only used by `TreeNode.clone`